### PR TITLE
Make imports from `astropy.units` and `astropy.constants` more consistent

### PR DESCRIPTION
**Pull request by namurphy**
This PR enables the [`pre-commit-search-and-replace`](https://github.com/mattlqx/pre-commit-search-and-replace) hook.

This hook makes these replacements:

 - `from astropy import units as u` → `import astropy.units as u`
 - `from astropy import constants as const` → `import astropy.constants as const`

I also added some common typos for this.

This PR looks big, but the main things to check are `.pre-commit-config.yaml` and `.pre-commit-search-and-replace.yaml`

### CODE REVIEW

File: .pre-commit-search-and-replace.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- search: from astropy import units as u
+  replacement: import astropy.units as u
+- search: from astropy import constants as const
+  replacement: import astropy.constants as const
+- search: Plasmapy|PlasmPy|Plasmpy|Palsmapy
+  replacement: PlasmaPy
```
Comment (namurphy) [inline .pre-commit-search-and-replace.yaml:6]: ...to catch common typos that have snuck in before.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,6 +42,12 @@ repos:
   - id: absolufy-imports
     exclude: docs/plasmapy_sphinx
 
+- repo: https://github.com/mattlqx/pre-commit-search-and-replace
+  rev: v1.0.5
+  hooks:
+  - id: search-and-replace
+    exclude: changelog/.*|docs/changelog/.*
```
Comment (namurphy) [inline .pre-commit-config.yaml:49]: This uses regular expressions, where the `.*` means any character, any number of times; and the `|` means "or".

File: .pre-commit-search-and-replace.yaml (hunk 1 above)
Comment (ejohnson-96) [reply .pre-commit-search-and-replace.yaml:6]: Good Idea 👍

**Review by ejohnson-96 — APPROVED** [review]
👍

**Pull request merged**